bin/BuildPackages.sh

```diff
@@ -8,26 +8,45 @@ mkdir -p "$LOGDIR"
 LOGFILE=buildpackages
 FAILPKGFILE=fail
 
-# print notices in green
-notice() {
-    printf "\033[32m%s\033[0m\n" "$@"
-}
-
-# print warnings in yellow
-warning() {
-    printf "\033[33mWARNING: %s\033[0m\n" "$@"
-}
-
-# print error in red and exit
-error() {
-    printf "\033[31mERROR: %s\033[0m\n" "$@"
-    exit 1
-}
-
-# print stderr error in red but do not exit
-std_error() {
-    printf "\033[31mERROR: %s\033[0m\n" "$@"
-}
+if [ -t 1 ]; then
```

Comment:
Come to think of it, I am not entirely convinced that even if the ````if```` is within the functions, it would work as intended, because the output is redirected using ````tee```` _after_ any of these commands are applied. I think to properly separate colouring to terminal and not colouring to files would involve checking everything that we write to terminal and one by one separate these two cases. 

I would opt for a simple check for ````--no-color````